src/sentry/integrations/vsts/integration.py

```diff
@@ -289,10 +291,18 @@ def build_integration(self, state):
         }
 
         try:
-            assert 'subscription' in IntegrationModel.objects.get(
+            integration_model = IntegrationModel.objects.get(
                 provider='vsts',
-                external_id=account['AccountId']
-            ).metadata
+                external_id=account['AccountId'],
+                status=ObjectStatus.VISIBLE,
+            )
+            assert 'subscription' in integration_model.metadata
+            org_integrations = OrganizationIntegration.objects.filter(
+                integration_id=integration_model.id,
+                status=ObjectStatus.VISIBLE,
+            ).exists()
```

Comment:
We do not update the Integration status or delete the integration at this time. Right now, the only way to tell if an integration has been "deleted" is to check that there are no organization_integrations associated with it. So I am currently checking this. It would be better in my opinion to change/implement deletion for integrations, but this is using what we have for now.

Reply:
Does using `filter(...).exists()` have any advantage over `get(...)`?

Reply:
Also, if you're using `exists()` here, can you just combine this and the next line into

```python
assert OrganizationIntegration.objects.filter(...).exists()
```

Reply:
So yes `exists` does in fact. The thing about `get` is that it executes immediately. Filters on the other hand are lazy. By calling exist, it'll just find the first one and let me know if it found it.  https://django.readthedocs.io/en/1.6.x/ref/models/querysets.html?highlight=exists#django.db.models.query.QuerySet.exists

https://django.readthedocs.io/en/1.6.x/topics/db/optimization.html?highlight=exists

Reply:
I don't know if it matters in this case – you're executing the query immediately either way. Makes sense, though.

Reply:
`exists` is a lighter weight query than get. `get` returns the whole model and in turn does basically a `SELECT *`. exists on the otherhand returns a boolean and does `SELECT 1` and doesn't even return back row data.

This is further optimized if there is actually a match of more than 1 row, exists due to the fact that it's `SELECT 1` will exit the query early. Whereas with `get` there's no LIMIT applied, and will return all rows and wait to raise a MultipleObjectsReturned error.

So `exists` by far is the most optimized way to do a check like this.